paasta_tools/vitesscluster_tools.py

```diff
@@ -15,6 +17,594 @@
+        "extraVolumeMounts": [
+            {
+                "mountPath": "/etc/vault/all_cas",
+                "name": "vault-secrets",
+                "readOnly": True,
+            },
+            {
+                "mountPath": "/etc/vitess_keyspace_acls",
+                "name": "acls",
+                "readOnly": True,
+            },
+            {
+                "mountPath": "etc/credentials.yaml",
+                "name": "vttablet-fake-credentials",
+                "readOnly": True,
+            },
+            {
+                "mountPath": "/etc/init_db.sql",
+                "name": "keyspace-fake-init-script",
+                "readOnly": True,
+            },
+        ],
+        "extraVolumes": [
+            {"name": "vault-secrets", "hostPath": {"path": "/nail/etc/vault/all_cas"}},
+            {
+                "name": "acls",
+                "hostPath": {"path": "/nail/srv/configs/vitess_keyspace_acls"},
+            },
+            {"name": "vttablet-fake-credentials", "hostPath": {"path": "/dev/null"}},
+            {"name": "keyspace-fake-init-script", "hostPath": {"path": "/dev/null"}},
+        ],
+        "replicas": 1,
+        "vttablet": {
+            "extraFlags": vttablet_extra_flags,
+            "resources": {
+                "requests": {
+                    "cpu": "100m",
+                    "memory": "256Mi",
+                },
+                "limits": {"memory": "256Mi"},
+            },
+        },
+        "externalDatastore": {
+            "database": db_name,
+            "host": SOURCE_DB_HOST,
+            "port": port,
+            "user": "vt_app",
+            "credentialsSecret": {
+                "key": "/etc/credentials.yaml",
+                "volumeName": "vttablet-fake-credentials",
+            },
+        },
+        "dataVolumeClaimTemplate": {
+            "accessModes": ["ReadWriteOnce"],
+            "resources": {"requests": {"storage": "10Gi"}},
+            "storageClassName": "ebs-csi-gp3",
+        },
+    }
+
+    vttablet_extra_env = copy.deepcopy(VTTABLET_EXTRA_ENV)
+    for env in vttablet_extra_env:
+        if env["name"] == "TOPOLOGY_FLAGS":
+            env[
+                "value"
+            ] = f"--topo_implementation {TOPO_IMPLEMENTATION} --topo_global_server_address ${zk_address} --topo_global_root {TOPO_GLOBAL_ROOT}"
+        if env["name"] == "CELL_TOPOLOGY_SERVERS":
+            env["value"] = zk_address
+        if env["name"] == "DB":
+            env["value"] = db_name
+        if env["name"] == "KEYSPACE":
+            env["value"] = keyspace
+        if env["name"] == "VAULT_ADDR":
+            env["value"] = f"https://vault-dre.{region}.yelpcorp.com:8200"
+        if env["name"] == "VAULT_CACERT":
+            env["value"] = f"/etc/vault/all_cas/acm-privateca-{region}.crt"
+        config["extraEnv"].append(env)
+
+    # Add extra pod label to filter
+    config["extraLabels"]["tablet_type"] = f"{db_name}_{tablet_type}"
```

Comment:
what's the longest db name and longest tablet type? labels are limited to 63 characters, so we'll likely need to truncate here (`limit_size_with_hash()` may be helpful for y'all here :))

Reply:
Oh, that's helpful to know! There are definitely some longer service names, and the restriction on MySQL side is only on the username which we shorten it based on db name https://yelpwiki.yelpcorp.com/pages/viewpage.action?pageId=63881224#HowTo:CreateanewMySQLServiceDatabase-Guidelines. Will use the function to shorten the label name accordingly